addon_imps/storage/s3.py

```diff
@@ -0,0 +1,112 @@
+# import dataclasses
+# import functools
+# import typing
+import boto3
+from asgiref.sync import sync_to_async
+from botocore import exceptions
+from django.core.exceptions import ValidationError
+
+# from addon_toolkit.cursor import OffsetCursor
+from addon_toolkit.interfaces import storage
+
+
+# from addon_service.authorized_storage_account.models import AuthorizedStorageAccount
+# from addon_service.credentials.models import ExternalCredentials
+
+
+class Boto3Client:
+    pass
+
+
+class S3ClientImp(storage.ClientRequestorImp):
+    client: Boto3Client
+
```

Comment:
My instinct is that either this should define `construct_client` or else the `client` type statement should just live in the S3StorageImp